homeassistant/components/unifiprotect/services.py

```diff
@@ -122,6 +150,29 @@ async def set_default_doorbell_text(hass: HomeAssistant, call: ServiceCall) -> N
     await _async_service_call_nvr(hass, call, "set_default_doorbell_message", message)
 
 
+async def remove_privacy_zone(hass: HomeAssistant, call: ServiceCall) -> None:
+    """Remove privacy zone from camera."""
+
+    name: str = call.data[ATTR_NAME]
+    camera = _async_get_ufp_camera(hass, call)
+
+    remove_index: int | None = None
+    for index, zone in enumerate(camera.privacy_zones):
+        if zone.name == name:
+            remove_index = index
+            break
+
+    if remove_index is None:
+        raise HomeAssistantError(
```

Comment:
https://developers.home-assistant.io/docs/core/platform/raising_exceptions?_highlight=raises

I think we want `ServiceValidationError` here